ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/utils/Keys.kt

```diff
@@ -0,0 +1,37 @@
+/**
+ * Contains functions to determine the type of key during encoding and writing.
+ */
+
+package com.akuleshov7.ktoml.utils
+
+/**
+ * Returns true if the specified key is a bare key, a key containing only
+ * alphanumeric, underscore, and hyphen characters.
+ *
+ * @return `true` if this key is a bare key.
+ */
+internal fun String.isBareKey() =
+        all {
+            it in 'A'..'Z' ||
+                    it in 'a'..'z' ||
+                    it in '0'..'9' ||
+                    it in "-_"
```

Comment:
Btw - why "-_" here? Why two symbols? I thought that we need to check only underscore

Reply:
Well, in the regex we did it too. A bare key can contain a hyphen

```kotlin
internal val bareKeyRegex = Regex("[A-Za-z0-9_-]+")
```

Reply:
ok, got it